### ISSUE TITLE
Extract partition list from pyarrow.dataset.ParquetFileFragment object

### ISSUE THREAD

**Issue by ayouqi** [apache/arrow#34212]
### Describe the enhancement requested

A` pyarrow.dataset.ParquetFileFragment` object has two elements: `path` and `partition`. The path can be extracted using `path` attribute but there is no attribute to get the partition.
For example for the following object:
`<pyarrow.dataset.ParquetFileFragment path=pq-test/Location=US/Industry=HT/dce9900c46f94ec3a8dca094cf62bd34-0.parquet partition=[Industry=HT, Location=US]>`

`object.path `returns `pq-test/Location=US/Industry=HT/dce9900c46f94ec3a8dca094cf62bd34-0.parquet` but `object.partition` has not been defined.

Also, it seems the order of partition list is reverse. 

### Component(s)

Python

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.

**Comment by ayouqi** [comment]
Any update?

**Comment by thisisnic** [comment]
CC @AlenkaF

**Comment by AlenkaF** [comment]
Thank you for the ping @thisisnic, this issue slipped our radar.

I think adding `partition` property to `FileFragment` would be a nice improvement.

For the time being, there is a possible workaround. As`ParquetFileFragment` is inheriting the [repr method from `FileFragment`](https://github.com/apache/arrow/blob/2a526c1e623df0ce8b9b76b03d4d9c617d21fda1/python/pyarrow/_dataset.pyx#L1982-L1995) and this method is using [`get_partition_keys`](https://github.com/apache/arrow/blob/80db1020881f461af3a300653cfd2333ef10a45e/python/pyarrow/_dataset.pyx#L4043) one could do the same on a `ParquetFileFragment.partition_expression` which is inherited from the Fragment class:

https://github.com/apache/arrow/blob/2a526c1e623df0ce8b9b76b03d4d9c617d21fda1/python/pyarrow/_dataset.pyx#L1480-L1484

Which would look similar to:

```python
get_partition_keys(ParquetFileFragment.partition_expression)
```

As for the ordering, it looks like it is non-deterministic (test checks both possible orderings)
https://github.com/apache/arrow/blob/2a526c1e623df0ce8b9b76b03d4d9c617d21fda1/python/pyarrow/tests/test_dataset.py#L1934-L1940

Summing up things that could be done here:
- add `partition` property
- better document `get_partition_keys` (add info to the User Guide, not only API docs)
- maybe document the non-deterministic ordering in the repr
